IceCream/Classes/CreamAsset.swift

```diff
@@ -31,16 +31,33 @@ public class CreamAsset: Object {
         save(data: data, to: uniqueFileName, shouldOverwrite: shouldOverwrite)
     }
 
-    private convenience init?(objectID: String, propName: String, url: URL) {
+    private convenience init?(objectID: String, propName: String, url: URL, copyFile: Bool) {
         guard let data = try? Data(contentsOf: url) else { return nil }
-        self.init(objectID: objectID, propName: propName, data: data)
-    }
 
+        if copyFile {
+            self.init()
+            self.uniqueFileName = "\(objectID)_\(propName).\(url.pathExtension)"
```

Comment:
In my specific case, the user can import any kind of files, and has the ability to preview them with Apple's QuickLook Preview framework. The problem is that if you pass a url without an extension, QLPreview doesn't know what to do with it and just reports it as a data file...  While with the extension it will properly display image/.pdf/.xls/.doc/.ppt/etc...

If there was no real extension to start with, I don't see the harm of appending that "01dca0f8e2750033506cc3b5e0b4c3bc319421519d"... ;)

Reply:
OK, I get you... How about we making another PR to fix the extension issue and discuss there :)

Reply:
Well I found that Swift supports symbolic link, so I can add back my extension just fine.

OTOH, copyItem will throw if the file already exists, thus return a nil asset when synch'ing from iCloud, on a device that already have those assets... ;)